javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -81,6 +120,275 @@ class Properties(protected val defaults: Properties)
       defaults.foreachAncestor(f)
   }
 
-  // def list(out: PrintStream): Unit
-  // def list(out: PrintWriter): Unit
+  private final val listStr = "-- listing properties --"
+
+  private def format(entry: ju.Map.Entry[AnyRef, AnyRef]): String = {
+    def format(s: String): String =
+      if (s.length > 40) s"${s.substring(0, 37)}..." else s
+
+    val key: String = entry.getKey().asInstanceOf[String]
+    val value: String = entry.getValue().asInstanceOf[String]
+
+    s"${key}=${format(value)}"
+  }
+
+  def list(out: PrintStream): Unit = {
+    out.println(listStr)
+    entrySet().scalaOps.foreach { entry => out.println(format(entry)) }
+  }
+
+  def list(out: PrintWriter): Unit = {
+    out.println(listStr)
+    entrySet().scalaOps.foreach { entry => out.println(format(entry)) }
+  }
+
+  private def loadImpl(reader: Reader): Unit = {
+    import java.util.regex._
+    val trailingBackspace = Pattern.compile("""(\\)+$""")
+    val br = new BufferedReader(reader)
+    val valBuf = new jl.StringBuilder()
+    var prevValueContinue = false
+    var isKeyParsed = false
+    var key: String = null
+    var line: String = null
+
+    while ({ line = br.readLine(); line != null }) {
+      var i: Int = -1
+      var ch: Char = Char.MinValue
+
+      def getNextChar(): Char = {
+        i += 1
+        // avoid out of bounds if value is empty
+        if (i < line.length())
+          line.charAt(i)
+        else
+          ch
+      }
+
+      def parseUnicodeEscape(): Char = {
+        val sb = new jl.StringBuilder()
+        var j = 0
+        while (j < 4) {
+          sb.append(line.charAt(i))
+          if (j < 3) {
+            // don't advance past the last char used
+            i += 1
+          }
+          j += 1
+        }
+        val ch = Integer.parseInt(sb.toString(), 16).toChar
+        ch
+      }
+
+      def isWhitespace(char: Char): Boolean =
+        char == ' ' || char == '\t' || char == '\f'
+
+      def isTokenKeySeparator(char: Char): Boolean =
+        char == '=' || char == ':'
+
+      def isKeySeparator(char: Char): Boolean =
+        isTokenKeySeparator(char) || isWhitespace(char)
+
+      def isEmpty(): Boolean =
+        line.isEmpty()
+
+      def isComment(): Boolean =
+        line.startsWith("#") || line.startsWith("!")
+
+      def oddBackslash(): Boolean = {
+        val m = trailingBackspace.matcher(line)
+        if (m.find()) {
+          val num = m.end(1) - m.start()
+          val isOdd = num % 2 != 0
+          isOdd
+        } else {
+          false
+        }
```

Comment:
Depending on `Pattern` and the entire regex machinery just to count the number of trailing backslashes is overkill. This method could be written just as easily without regexes. Something like
```scala
var i = line.length()
while (i > 0 && line.charAt(i - 1) == '\\')
  i -= 1
(line.length() - i) % 2 != 0
```
gosh, it's even shorter.

(I may have overlooked this in Scala Native; I'm more acutely aware of code size and hence dependencies on heavy components inside Scala.js)

Reply:
This is very slick solution.